pubsub.go

```diff
@@ -329,15 +364,35 @@ func (p *PubSub) processLoop(ctx context.Context) {
 				messages <- p.getHelloPacket()
 				go p.handleNewPeer(ctx, pid, messages)
 				p.peers[pid] = messages
+				p.inflightPeers[pid] = struct{}{}
 				continue
 			}
 
 			delete(p.peers, pid)
+			delete(p.inflightPeers, pid)
 			for _, t := range p.topics {
 				delete(t, pid)
 			}
 
 			p.rt.RemovePeer(pid)
+			p.notifyNewPeer(pid)
+
+		case add := <-p.addPeerNotif:
```

Comment:
this logic is completely redundant, we already handle this with connection notifications.

Reply:
I am not sure I understand where the redundancy comes in. Are you saying `notifyNewPeer` there will never convey information not handled in `notifyNewPeer` elsewhere (e.g. in the newPeerStream and newPeerErr channels)? Or is this a higher level "general redundancy"? Either way more explanation would help me understand.

Reply:
No we shouldn't have to care about in flight peers and all the details of discovery, this leaks through the abstraction barrier.
The only thing we care about is whether bootstrap is complete or not and that can be handled with a single event in the pubsub mainloop.